### PR TITLE
Add modern authentication auth class

### PR DESCRIPTION
**Pull request by kevincarrogan**
This adds an authentication class for OAuth based authentication for pop3 connections.

At this point in time this is unused but is laying the ground work for the switch to OAuth.

### CODE REVIEW

File: mail/auth.py [hunk 1]
```diff
@@ -21,3 +25,57 @@ def authenticate(self, connection: poplib.POP3_SSL):
 
     def __eq__(self, other: Authenticator):
         return self.user == other.user and self.password == other.password
+
+
+class ModernAuthentication:
```
Comment (currycoder) [inline mail/auth.py:30]: Out of interest, how have you been testing this end to end? I assume that you have test creds and point to an integration env etc?  Could it be worth documenting that somewhere?

Comment (currycoder) [reply mail/auth.py:30]: And I take it that your testing confirmed you could read emails from the mailbox?

File: mail/tests/test_auth.py [hunk 2]
```diff
@@ -27,3 +28,141 @@ def test_not_equal(self):
         equal_auth = BasicAuthentication("diff_user", "diff_password")
 
         self.assertNotEqual(auth, equal_auth)
+
+
+class ModernAuthenticationTests(SimpleTestCase):
+    def test_authenticates_connection_with_silent_acquisition(self):
```
Comment (currycoder) [inline mail/tests/test_auth.py:34]: I don't really know the difference between "silent acquisition" and non-silent...  Could some docstrings be added to these tests to help explain?

File: mail/auth.py (hunk 1 above)
Comment (kevincarrogan) [reply mail/auth.py:30]: Yes. This has all been tested with real credentials, however that whole thing is setup and dealt with by SRE and I've just been handed those credentials.

Comment (kevincarrogan) [reply mail/auth.py:30]: They will eventually end up in vault and we'll pull from there when I tie this all up with the rest of the code.

**Review by currycoder — APPROVED** [review]

**Pull request merged**